modules/markup_field/src/Plugin/Field/FieldType/MarkupFieldItem.php

```diff
@@ -0,0 +1,290 @@
+ *   default_formatter = "rendered_markup",
+ *   category = @Translation("Field"),
+ * )
+ */
+class MarkupFieldItem extends FieldItemBase {
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * The renderer.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * The library discovery service.
+   *
+   * @var \Drupal\Core\Asset\LibraryDiscoveryInterface
+   */
+  protected $libraryDiscovery;
+
+  /**
+   * The library dependency resolver.
+   *
+   * @var \Drupal\Core\Asset\LibraryDependencyResolverInterface
+   */
+  protected $libraryDependencyResolver;
+
+  /**
+   * The CSS asset collection renderer service.
+   *
+   * @var \Drupal\Core\Asset\AssetCollectionRendererInterface
+   */
+  protected $cssCollectionRenderer;
+
+  /**
+   * The JS asset collection renderer service.
+   *
+   * @var \Drupal\Core\Asset\AssetCollectionRendererInterface
+   */
+  protected $jsCollectionRenderer;
+
+  /**
+   * The state key value store.
+   *
+   * @var \Drupal\Core\State\State
+   */
+  protected $state;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct($definition, $name, TraversableTypedDataInterface $parent, EntityFieldManagerInterface $entity_field_manager, RendererInterface $renderer, LibraryDiscoveryInterface $library_discovery, LibraryDependencyResolverInterface $library_dependency_resolver, AssetCollectionRendererInterface $css_collection_renderer, AssetCollectionRendererInterface $js_collection_renderer, State $state) {
+    parent::__construct($definition, $name, $parent);
+    $this->entityFieldManager = $entity_field_manager;
+    $this->renderer = $renderer;
+    $this->libraryDiscovery = $library_discovery;
+    $this->libraryDependencyResolver = $library_dependency_resolver;
+    $this->cssCollectionRenderer = $css_collection_renderer;
+    $this->jsCollectionRenderer = $js_collection_renderer;
+    $this->state = $state;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function createInstance($definition, $name = NULL, TraversableTypedDataInterface $parent = NULL) {
+    return new static(
+      $definition,
+      $name,
+      $parent,
+      \Drupal::service('entity_field.manager'),
+      \Drupal::service('renderer'),
+      \Drupal::service('library.discovery'),
+      \Drupal::service('library.dependency_resolver'),
+      \Drupal::service('asset.js.collection_renderer'),
+      \Drupal::service('asset.css.collection_renderer'),
+      \Drupal::service('state')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function propertyDefinitions(FieldStorageDefinitionInterface $field_definition) {
+    $properties['value'] = DataDefinition::create('string')
+      ->setLabel(t('Field markup'))
+      ->setComputed(TRUE);
+    $properties['assets'] = MapDataDefinition::create()
+      ->setLabel(t('Assets'))
+      ->setComputed(TRUE);
```

Comment:
this is not computed, should be removed.

Reply:
Please add a description to clarify this is an array in the format as taken by #attached render arrays.